### PR TITLE
[Android] Update methods to check if the headset is plugged in

### PR DESCRIPTION
**Pull request by joseluismarti**
## Description
The methods  ```isWiredHeadsetOn()``` and ```isBluetoothA2dpOn()``` used to identify whether a headset is plugged in are deprecated.

As suggested in the [documentation](https://developer.android.com/reference/android/media/AudioManager#getDevices(int)), we should use the ```getDevices()``` method instead to get a list of all the audio devices currently connected to the system.

This PR and #20019 complete the update of all the deprecated methods of the AudioManager class.

## How has this been tested?
I've checked that the new method returns the correct value when plugging in or unplugging bluetooth and wired headsets.

Tested on Android 11 phone

## What is the effect on users?
No direct effect to the user

## Types of change
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/platform/android/activity/XBMCApp.h [hunk 1]
```diff
@@ -276,6 +276,8 @@ class CXBMCApp : public IActivityHandler,
   int64_t m_frameTimeNanos{0};
   float m_refreshRate{0.0f};
 
+  bool isHeadsetPlugged();
```
Comment (ksooo) [inline xbmc/platform/android/activity/XBMCApp.h:279]: There is no need to expose this method in the header as it does not use any members of CXBMCApp. Please make it an internal helper function.

File: xbmc/platform/android/activity/XBMCApp.cpp [hunk 2]
```diff
@@ -1587,3 +1586,32 @@ void CXBMCApp::surfaceDestroyed(CJNISurfaceHolder holder)
 
   m_window.reset();
 }
+
+bool CXBMCApp::isHeadsetPlugged()
```
Comment (ksooo) [inline xbmc/platform/android/activity/XBMCApp.cpp:1590]: namespace
{
bool IsHeadsetPlugged()
{
...
}
} // namespace

File: xbmc/platform/android/activity/XBMCApp.cpp [hunk 3]
```diff
@@ -1587,3 +1586,32 @@ void CXBMCApp::surfaceDestroyed(CJNISurfaceHolder holder)
...
+  CJNIAudioManager audioManager(getSystemService("audio"));
+
+  if (CJNIBuild::SDK_INT >= 26)
+  {
+    const CJNIAudioDeviceInfos devices = 
+        audioManager.getDevices(CJNIAudioManager::GET_DEVICES_OUTPUTS);
+
+    for (const CJNIAudioDeviceInfo& device : devices)
+    {
+      int type = device.getType();
```
Comment (ksooo) [inline xbmc/platform/android/activity/XBMCApp.cpp:1601]: `const int`

**Review by ksooo — CHANGES_REQUESTED** [review]

File: xbmc/platform/android/activity/XBMCApp.cpp (hunk 2 above)
Comment (joseluismarti) [reply xbmc/platform/android/activity/XBMCApp.cpp]: ok, move to an unnamed namespace

**Review by ksooo — APPROVED** [review]

**Pull request merged**